transport/server.cc

```diff
@@ -207,11 +207,68 @@ cql_server::cql_server(distributed<cql3::query_processor>& qp, auth::service& au
+                        sm::description("Counts the total number of raised CQL UNAVAILABLE errors.")),
+
+        sm::make_derive("overloaded_errors", _stats.errors[exceptions::exception_code::OVERLOADED],
+                        sm::description("Counts the total number of raised CQL OVERLOADED errors.")),
+
+        sm::make_derive("bootstrapping_errors", _stats.errors[exceptions::exception_code::IS_BOOTSTRAPPING],
+                        sm::description("Counts the total number of raised CQL IS_BOOTSTRAPPING errors.")),
+
+        sm::make_derive("truncate_errors", _stats.errors[exceptions::exception_code::TRUNCATE_ERROR],
+                        sm::description("Counts the total number of raised CQL TRUNCATE errors.")),
+
+        sm::make_derive("write_timeouts", _stats.errors[exceptions::exception_code::WRITE_TIMEOUT],
+                        sm::description("Counts the total number of raised CQL WRITE_TIMEOUT errors.")),
+
+        sm::make_derive("read_timeouts", _stats.errors[exceptions::exception_code::READ_TIMEOUT],
+                        sm::description("Counts the total number of raised CQL READ_TIMEOUT errors.")),
+
+        sm::make_derive("read_failures", _stats.errors[exceptions::exception_code::READ_FAILURE],
+                        sm::description("Counts the total number of raised CQL READ_FAILURE errors.")),
 
+        sm::make_derive("function_failures", _stats.errors[exceptions::exception_code::FUNCTION_FAILURE],
+                        sm::description("Counts the total number of raised CQL FUNCTION_FAILURE errors.")),
+
+        sm::make_derive("write_failures", _stats.errors[exceptions::exception_code::WRITE_FAILURE],
+                        sm::description("Counts the total number of raised CQL WRITE_FAILURE errors.")),
+
+        sm::make_derive("cdc_write_failures", _stats.errors[exceptions::exception_code::CDC_WRITE_FAILURE],
+                        sm::description("Counts the total number of raised CQL CDC_WRITE_FAILURE errors.")),
+
+        sm::make_derive("syntax_errors", _stats.errors[exceptions::exception_code::SYNTAX_ERROR],
+                        sm::description("Counts the total number of raised CQL SYNTAX errors.")),
+
+        sm::make_derive("unathorized_errors", _stats.errors[exceptions::exception_code::UNAUTHORIZED],
+                        sm::description("Counts the total number of raised CQL UNATHORIZED errors.")),
+
+        sm::make_derive("invalid_errors", _stats.errors[exceptions::exception_code::INVALID],
+                        sm::description("Counts the total number of raised CQL INVALID errors.")),
+
+        sm::make_derive("config_errors", _stats.errors[exceptions::exception_code::CONFIG_ERROR],
+                        sm::description("Counts the total number of raised CQL CONFIG errors.")),
+
+        sm::make_derive("already_exists_errors", _stats.errors[exceptions::exception_code::ALREADY_EXISTS],
```

Comment:
[] can throw. Will probably wreak havoc on the metric collector.

Reply:
It's a small map, how likely are we to actually get something thrown out of there?
I'm now looking at labels to see if that changes what I'm using to keep track of the errors, but if not, what would you suggest I do instead? List each stat as separate member?